### ISSUE TITLE
Composer path repository always mirroring on Windows

### ISSUE THREAD

**Issue by nebulousGirl** [composer/composer#4695]
When I run `composer update`, local packages always get mirrored. 

I have tried creating the symlink myself using PHP's `symlink` function and it worked, my package is symlinked and composer recognizes it.

My question is why composer does not symlink when installing? Is there a way to debug this. Right now, all I get from composer is `Mirrored from ...`

Here is my configuration: 
```json
        {
            "type": "path",
            "url": "../shared/Pages"
        }
```

**Comment by alcohol** [comment]
Afaik Windows does not handle relative symlinks very well at the moment (#4690). But without more information I'm not really sure (I don't have any Windows machines that I develop on).

Could you run an install or update with `-vvv` and share the parts that are relevant to this path repo?

**Comment by curry684** [comment]
> I have tried creating the symlink myself

What privileges did you try this under? I've just tested this to **only** work in an elevated account.

**Comment by nebulousGirl** [comment]
I am on windows 10, using wampserver. I used the PHP `symlink` function from my project codebase and it worked.

Now, that I think about it. The cli version of PHP might not have the same privileges as the one running under Apache.

**Comment by curry684** [comment]
Can you check in `services.msc` (or Task Manager) which account the Apache process is running under?

**Comment by nebulousGirl** [comment]
The user name for httpd.exe is `SYSTEM` while my command line processes are running under my Windows account name.

**Comment by curry684** [comment]
`SYSTEM` most definitely has privileges to make symlinks indeed, and is not limited by UAC.

You can run Composer from an elevated command prompt (right-click command prompt and 'Run as Administrator'), but that's *not really a smart idea* to say the least. Remember that Composer is downloading program code capable of killing your system.

**Comment by nebulousGirl** [comment]
I tried running as an admin earlier, just to check if it was working (I know that it is not safe), but it didn't work either.

I think I will have to find a workaround or just continue to use `symlink` manually during the first installation.

Thanks for the feedback. At least, I know why I was able to create symlinks while Composer was not.

**Comment by curry684** [comment]
I think @alcohol is also correct with relative symlinks not working all that well under Windows anyway, but my guess is that the combination of both means it's not going to be fixed anywhere soon hehe.

**Comment by curry684** [comment]
Are you able to test whether #4845 fixes your issues?

**Comment by Seldaek** [comment]
Fixed by #4845
